magma/backend/coreir/coreir_utils.py

```diff
@@ -146,6 +152,8 @@ def map_genarg(context, value):
         if value.is_input():
             return context.named_types[("coreir", "arstIn")]
         return context.named_types[("coreir", "arst")]
+    if isinstance(value, Kind):
```

Comment:
is this vestigial from the primitives we're no longer using?

Reply:
Yea, but it's useful for having a coreir genarg parameter `T` (e.g. Wire(T)), but since we may be dropping support for this backend not sure if we need to worry about testing/maintenance

Reply:
ok, think I'd prefer pulling that out into a different change unless it's otherwise needed for this change. I can make an issue for it.